src/bootstrap/install.rs

```diff
@@ -72,8 +72,11 @@ fn install_sh(
     let bindir_default = PathBuf::from("bin");
     let libdir_default = PathBuf::from("lib");
     let mandir_default = datadir_default.join("man");
-    let prefix = build.config.prefix.as_ref().unwrap_or(&prefix_default);
-    let sysconfdir = build.config.sysconfdir.as_ref().unwrap_or(&sysconfdir_default);
+    let canonicalize = |p: &PathBuf| {
+        fs::canonicalize(p).expect(&format!("could not canonicalize {}", p.display()))
+    };
+    let prefix = build.config.prefix.as_ref().map_or(prefix_default, canonicalize);
+    let sysconfdir = build.config.sysconfdir.as_ref().map_or(sysconfdir_default, canonicalize);
```

Comment:
I think we might have to be more careful about `sysconfdir` here. If it's an absolute path, then canonicalizing it is fine, but if it's relative, it'll be relative to `prefix` per the documentation, which makes this code (I believe) do the wrong thing.

I think I'd probably branch on is_absolute/is_relative on the path after mapping it to the default to determine if we need to canonicalize it (though it might be true that like all of the other paths, there's never a need to do so?).

Reply:
Oh, you're absolutely right. I was confused by the default value being `"/etc"` rather than "etc". Fixed.